Client/Frontend/Home/HomeTabBanner.swift

```diff
@@ -123,54 +130,103 @@ class DefaultBrowserCard: UIView {
-            closeButton.widthAnchor.constraint(equalToConstant: 16),
+            dismissButton.topAnchor.constraint(equalTo: cardView.topAnchor, constant: 16),
+            dismissButton.trailingAnchor.constraint(equalTo: cardView.trailingAnchor, constant: -16),
+            dismissButton.heightAnchor.constraint(equalToConstant: 16),
+            dismissButton.widthAnchor.constraint(equalToConstant: 16),
 
-            learnHowButton.centerXAnchor.constraint(equalTo: cardView.centerXAnchor),
-            learnHowButton.bottomAnchor.constraint(equalTo: cardView.bottomAnchor, constant: -16),
-            learnHowButton.widthAnchor.constraint(equalToConstant: DefaultBrowserCardUX.learnHowButtonSize.width),
-            learnHowButton.heightAnchor.constraint(equalToConstant: DefaultBrowserCardUX.learnHowButtonSize.height)
+            ctaButton.centerXAnchor.constraint(equalTo: cardView.centerXAnchor),
+            ctaButton.bottomAnchor.constraint(equalTo: cardView.bottomAnchor, constant: -16),
+            ctaButton.widthAnchor.constraint(equalToConstant: UX.learnHowButtonSize.width),
+            ctaButton.heightAnchor.constraint(equalToConstant: UX.learnHowButtonSize.height)
         ])
     }
 
+    /// Apply message data, including handling of cases where certain parts of the message are missing.
+    private func applyMessage() {
+
+        /// If no messages exist, continue using our evergreen message.
+        guard let message = message else {
+            bannerTitle.text = String.DefaultBrowserCardTitle
+            descriptionText.text = String.DefaultBrowserCardDescription
+            ctaButton.setTitle(String.DefaultBrowserCardButton, for: .normal)
+
+            return
+        }
+
+        if let buttonLabel = message.data.buttonLabel {
+            ctaButton.setTitle(buttonLabel, for: .normal)
+        } else {
+            ctaButton.removeFromSuperview()
+            let cardTapped = UITapGestureRecognizer(target: self, action: #selector(handleCTA))
+
+            cardView.addGestureRecognizer(cardTapped)
+            cardView.isUserInteractionEnabled = true
+        }
+
+        if let title = message.data.title {
+            bannerTitle.text = title
+        } else {
+            cardView.willRemoveSubview(bannerTitle)
+            bannerTitle.removeFromSuperview()
+        }
+
+        descriptionText.text = message.data.text
+        messagingManager.onMessageDisplayed(message)
+    }
+
     @objc private func dismissCard() {
         self.dismissClosure?()
-        UserDefaults.standard.set(true, forKey: "DidDismissDefaultBrowserCard")
-        TelemetryWrapper.gleanRecordEvent(category: .action, method: .tap, object: .dismissDefaultBrowserCard)
+
+        guard let message = message else {
+            /// If we're here, that means we've shown the evergreen. Handle it as we always did.
+            UserDefaults.standard.set(true, forKey: "DidDismissDefaultBrowserCard")
```

Comment:
We sholud use `PrefsKeys.KeyDidShowDefaultBrowserOnboarding` for these strings? Check before, because it was being used as the string originally. WHYYYYYYyyyy?